transitions/core.py

```diff
@@ -497,24 +497,30 @@ def add_states(self, states, on_enter=None, on_exit=None,
                 state = State(**state)
             self.states[state.name] = state
             for model in self.models:
-                self._add_state_to_model(state, model)
+                self._add_model_to_state(state, model)
         # Add automatic transitions after all states have been created
         if self.auto_transitions:
             for s in self.states.keys():
                 self.add_transition('to_%s' % s, '*', s)
 
-    def _add_state_to_model(self, state, model):
+    def _add_model_to_state(self, state, model):
         setattr(model, 'is_%s' % state.name,
                 partial(self.is_state, state.name, model))
+
         #  Add enter/exit callbacks if there are existing bound methods
-        enter_callback = 'on_enter_' + state.name
-        if hasattr(model, enter_callback) and \
-                inspect.ismethod(getattr(model, enter_callback)):
-            state.add_callback('enter', enter_callback)
-        exit_callback = 'on_exit_' + state.name
-        if hasattr(model, exit_callback) and \
-                inspect.ismethod(getattr(model, exit_callback)):
-            state.add_callback('exit', exit_callback)
+        for trigger in ('enter', 'exit'):
+            try:
+                state.add_callback(trigger, self._lookup_callback(model, state, trigger))
```

Comment:
I had initially intended to implement a state.remove_callback, until I realized that this is adding a string and not an actual function callback (which would have created a reference) to the string.

Is there any harm in leaving stale callback registrations on the state object?